src/averageAges.js

```diff
@@ -52,7 +69,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothersNames = people.map(x => x.mother);
+  const mothers = people.filter(x => mothersNames.includes(x.name));
+  const peoples = onlyWithSon ? people.filter(x => x.sex === 'm') : [...people];
+  const result = [];
+
+  mothers.map(mother => {
```

Comment:
```suggestion
  const result = mothers.map(mother => {
```

map повертає масив, давай використаємо цю його особливість замість того щоб імітувати поведінку foreach і пушити елементи в якийсь масив ззвоні